chore: enhance .NET onboarding story (#15755)
microsoft/playwright: docs/src/test-runners-csharp.md

For CPU-bound tests, we recommend using as many workers as there are cores on your system, divided by 2. For IO-bound tests you can use as many workers as you have cores.

### Customizing [BrowserContext] options

> To override the context options, you can override the `ContextOptions` method inside the `Page` or `BrowserContext` base class which you get from `Microsoft.Playwright.NUnit`. See the following example:

Reviewer: ```suggestion
To customize context options, you can override the `ContextOptions` method of your test class derived from `Microsoft.Playwright.NUnit.PageTest` or `Microsoft.Playwright.MSTest.PageTest`. See the following example:
```

```csharp
using Microsoft.Playwright.NUnit;

namespace PlaywrightTests;

[Parallelizable(ParallelScope.Self)]
public class MyTest : PageTest
{
    [Test]
    public async Task TestWithCustomContextOptions()
    {
        // The following Page (and BrowserContext) instance has the custom colorScheme, viewport and baseURL set:
        await Page.GotoAsync("/login");
    }

    public override BrowserNewContextOptions ContextOptions()
    {
        return new BrowserNewContextOptions()
        {
            ColorScheme = ColorScheme.Light,
            ViewportSize = new()
            {
                Width = 1920,
                Height = 1080
            },
            BaseURL = "https://github.com",
        };
    }
}
```

### Customizing [Browser]/launch options

[Browser]/launch options can be override either using a run settings file or by setting the run settings options directly via the
CLI. See the following example:

```xml
<?xml version="1.0" encoding="utf-8"?>
<RunSettings>
  <TestRunParameters>

>     <Parameter name="browser" value="webkit" />

Reviewer: I don't think `webkit msedge` is going to fly :)

    <Parameter name="headless" value="false" />

>     <Parameter name="channel" value="msedge" />

Reviewer: We should list all available options, perhaps in this doc?

Author: lets follow up
